devtools/x/src/utils.rs

```diff
@@ -54,64 +46,101 @@ pub fn apply_sccache_if_possible(
                     project_root().to_str().unwrap_or_default(),
                     var_os("CARGO_HOME").unwrap_or_default().to_string_lossy()
                 );
-            } else {
-                if !install_if_needed(cargo_config, "sccache", &sccache_config.installer) {
-                    return Err(anyhow!("Failed to install sccache, bailing"));
-                }
-                stop_sccache_server();
-                envs.push(("RUSTC_WRAPPER", Some("sccache".to_owned())));
-                envs.push(("CARGO_INCREMENTAL", Some("false".to_owned())));
-                envs.push(("SCCACHE_BUCKET", Some(sccache_config.bucket.to_owned())));
-                if let Some(ssl) = &sccache_config.ssl {
-                    envs.push((
-                        "SCCACHE_S3_USE_SSL",
-                        if *ssl {
-                            Some("true".to_owned())
-                        } else {
-                            Some("false".to_owned())
-                        },
-                    ));
-                }
+            }
+            correct_location
+        } else {
+            false
+        }
+    } else {
+        false
+    }
+}
 
-                if let Some(url) = &sccache_config.endpoint {
-                    envs.push(("SCCACHE_ENDPOINT", Some(url.to_owned())));
-                }
+/// Logs the output of "sccache --show-stats"
+pub fn log_sccache_stats() {
+    info!("Sccache statistics:");
+    let mut sccache = Command::new("sccache");
+    sccache.arg("--show-stats");
+    sccache.stdout(Stdio::inherit()).stderr(Stdio::inherit());
+    if let Err(error) = sccache.output() {
+        warn!("Could not log sccache statistics: {}", error);
+    }
+}
 
-                if let Some(extra_envs) = &sccache_config.envs {
-                    for (key, value) in extra_envs {
-                        envs.push((key, Some(value.to_owned())));
-                    }
-                }
+pub fn stop_sccache_server() {
```

Comment:
Seems like you should warn if you fail to stop the server?

Reply:
added